dev/start

```diff
@@ -40,7 +40,7 @@ ENABLE_OIDC_OKTA=false
 ENABLE_ROTATORS=false
 IDENTITY_USER=""
 
-declare -A required_envvars
+declare -a required_envvars
 required_envvars[identity]="IDENTITY_CLIENT_ID IDENTITY_CLIENT_SECRET IDENTITY_PROVIDER_URI"
 required_envvars[azure]="AZURE_TENANT_ID AZURE_SUBSCRIPTION_ID AZURE_RESOURCE_GROUP"
 
```

Comment:
Not related to changes you've made here, but I noticed that `init_iam` never gets called in the dev environment, so the `--authn-iam` seemingly does nothing

Reply:
@gl-johnson I made this change because `dev/start` wouldn't start without this change.

Reply:
Great spot!  I've added it into the dev script.